processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -32,13 +32,13 @@
  * provided to the Aggregator through the MetricSelector object, so whatever creates that object gets to choose how
  * the data is actually stored and accessed.
  */
-public interface AggregatorFactory
+public abstract class AggregatorFactory
```

Comment:
let us keep it like this for now, unless someone else comes along arguing one way or the other :)